pkg/controller/controller.go

```diff
@@ -166,56 +181,87 @@ func (haproxy *HAProxyController) OnUpdate(cfg ingress.Configuration) error {
 		return nil
 	}
 
-	out, err := haproxy.reloadHaproxy()
+	reloadCmd := haproxy.reloadHaproxy(configFile)
+	out, err := reloadCmd.CombinedOutput()
 	if len(out) > 0 {
-		glog.Infof("HAProxy output:\n%v", string(out))
+		glog.Infof("HAProxy[pid=%v] output:\n%v", reloadCmd.Process.Pid, string(out))
 	}
 	return err
 }
 
 // RewriteConfigFiles safely replaces configuration files with new contents after validation
-func rewriteConfigFiles(data []byte, reloadStrategy, configFile string) error {
-	tmpf := "/etc/haproxy/new_cfg.erb"
-
-	err := ioutil.WriteFile(tmpf, data, 644)
-	if err != nil {
-		glog.Warningln("Error writing rendered template to file")
-		return err
+func (haproxy *HAProxyController) rewriteConfigFiles(data []byte) (string, error) {
+	// Include timestamp in config file name to aid troubleshooting. When using a single, ever-changing config file it
+	// was difficult to know what config was loaded by any given haproxy process
+	timestamp := time.Now().Format("-060102-150405.0000")
```

Comment:
What about four digits to the year and three to the milisseconds?

Reply:
@jcmoraisjr Do you mean a formate like `-0102-150405.000`, where year and nanoseconds are lost but month, day, hour, second, millisecond are retained? I think that will be fine since there is low risk of overwriting based on either of those.

Reply:
I mean following this convention: `-20060102-150405.000`.